extra/release_tool.py

```diff
@@ -670,7 +670,10 @@ def do_list_repos(args, optional_too, only_backend):
     repos_versions_dict = {}
     for repo in repos:
         if repo.name == "integration":
-            repos_versions_dict[repo.name] = args.in_integration_version
+            if args.in_integration_version:
+                repos_versions_dict[repo.name] = args.in_integration_version
+            else:
+                repos_versions_dict[repo.name] = "HEAD"
```

Comment:
Actually, I think a better approach here is to remove the `if repo.name == "integration"` entirely, and let "integration" follow the path into `version_of` which already deals with this in a smarter way. The algorithm is quite slow, but it figures out the real branch identity of HEAD instead of just printing it outright.

I just tried it locally, it works.

Reply:
Ok, thanks.

I'll try that a bit more in with different combinations because originally it did break, and then I added the special case handling fro integration.